### DISCUSSION TITLE
item cannot be resolved to a variable

### DISCUSSION THREAD

**Discussion by snappie321** [FabricMC/fabric-api#3053] [Mod Dev Support] [answered]
I'm getting this error, fixes on internet don't seem to work.

```
package net.testmod;

import org.apache.http.config.Registry;
import com.mojang.datafixers.TypeRewriteRule.One;

import net.fabricmc.api.ModInitializer;
import net.fabricmc.fabric.api.item.v1.FabricItemSettings;
import net.minecraft.item.Item;
import net.minecraft.item.ItemGroup;
import net.minecraft.registry.Registries;
import net.minecraft.util.Identifier;


public class Testmod implements ModInitializer {

    public static final Item CUSTOM_ITEM = new Item(new FabricItemSettings());

    @Override
    public void onInitialize() {
        Registry.register(Registries.ITEM, new Identifier("testmod", "fabric_item"), Ruby_item);
    }   
}
```

Its for 1.19.4 and im just trying out

**Comment by warjort** [comment, marked as answer]
This is very basic java. I suggest you buy a book on learning java or use an online course.

The only object you have defined of type `Item` you decided to call `CUSTOM_ITEM`
You are then trying to register an object of type `Item` called `Ruby_item` that you haven't defined anywhere.

You are also trying to use the Registry class related to apache's http client instead of Minecraft's Registry class.
-import org.apache.http.config.Registry;
+import net.minecraft.registry.Registry;
